### DISCUSSION TITLE
Suggestions on starting out.

### DISCUSSION THREAD

**Discussion by kaiuri** [dpploy/chen-3170#32] [General]
### About getting the tools

From the same initial assumptions of [00-syllabus.ipynb](https://github.com/dpploy/chen-3170/blob/master/notebooks/00-syllabus.ipynb) and to avoid those famous *dependency issues 🔥* related to setting up your environment, I'd recommend the following package managers as tools for the not-so-die-hard-programmers:

On **Windows**:
- [Chocolatey](https://chocolatey.org/)

On **macOS**:
- [Homebrew](https://brew.sh/)

On **Linux**:
- [Nixpkgs](https://gist.github.com/kaiuri/2d45834d5a0232c4f1a9b839ac127587) for a true sandbox approach.

#### Code Editors

Before hitting that awesome Jupyter code block, writing your code with a capable text editor can be very helpful as you'll be able to check your it for errors and keep up with Python's language formatting standards.

*Suggestions*:

1. Vim/Neovim: [bootstrap-vim]( https://vim-bootstrap.com/ ) can help you get started.
2. Emacs
3. Sublime Text
4. VSCode

If you intend on following up with programming, choosing Vim or Emacs will most surely help your productivity.

### About Anaconda

On line [74](https://github.com/dpploy/chen-3170/blob/5d74367ab95699cffc2a39da6f178de8bb57e78b/notebooks/00-syllabus.ipynb?short_path=75e247f#L74) of [00-syllabus.ipynb](https://github.com/dpploy/chen-3170/blob/master/notebooks/00-syllabus.ipynb), it's suggested the students install Anaconda as a way of getting the Python-Jupyter resources.
I'd like to differ and recommend the smaller version of Anaconda, **Miniconda**, which is also readily available at [https://conda.io](https://conda.io).

Miniconda is considerably smaller than Anaconda and the user can add packages as they're need, where Anaconda has a much larger prepackaged toolchain which most of the time goes on unused.

### On the matter of the resources available.

Google offers two very capable free options to the ones already mentioned in line [72](https://github.com/dpploy/chen-3170/blob/1ee24a83e88c6f0f32db3107df2a4f2748370f25/notebooks/00-syllabus.ipynb?short_path=75e247f#L72) of [00-syllabus.ipynb](https://github.com/dpploy/chen-3170/blob/master/notebooks/00-syllabus.ipynb).

* [**Colaboratory**](https://colab.research.google.com/),
* [**Kaggle**](https://www.kaggle.com/)

Which have the following attributes.
* Production and execution of arbitrary python code through the browser,
* Both are especially well suited for:
   - machine learning,
   - data analysis and
   - education.

Technically, those are hosted Jupyter notebook services requiring no initial setup to use while providing free access to Google's computing resources including  CPUs, TPUs and GPUs through the web browser.

**Additionally** one might run those notebooks locally through  the `jupyter_http_over_ws` available at PyPl.

**Comment by dealmeidavf** [comment]
Thank you Uriel. Sorry I did not monitor this.
